src/components/apis/api-products/ko/runtime/api-products-dropdown.ts

```diff
@@ -0,0 +1,144 @@
+import * as ko from "knockout";
+import * as Constants from "../../../../../constants";
+import template from "./api-products-dropdown.html";
+import { Component, RuntimeComponent, OnMounted, Param, OnDestroyed } from "@paperbits/common/ko/decorators";
+import { Router } from "@paperbits/common/routing";
+import { ApiService } from "../../../../../services/apiService";
+import { SearchQuery } from "../../../../../contracts/searchQuery";
+import { RouteHelper } from "../../../../../routing/routeHelper";
+import { Product } from "../../../../../models/product";
+
+@RuntimeComponent({
+  selector: "api-products-dropdown-runtime",
+})
+@Component({
+  selector: "api-products-dropdown-runtime",
+  template: template,
+})
+export class ApiProductsDropdown {
+  public readonly products: ko.ObservableArray<Product>;
+  public readonly selectedApiName: ko.Observable<string>;
+  public readonly working: ko.Observable<boolean>;
+  public readonly pattern: ko.Observable<string>;
+  public readonly pageNumber: ko.Observable<number>;
+  public readonly totalPages: ko.Observable<number>;
+  public readonly selectedProduct: ko.Observable<Product>;
+  public readonly selectedProductName: ko.Observable<string>;
+  public readonly selection: ko.Computed<string>;
+
+  constructor(
+    private readonly apiService: ApiService,
+    private readonly router: Router,
+    private readonly routeHelper: RouteHelper
+  ) {
+    this.detailsPageUrl = ko.observable();
+    this.allowSelection = ko.observable(false);
+    this.products = ko.observableArray([]);
+    this.selectedApiName = ko.observable();
+    this.working = ko.observable();
+    this.pattern = ko.observable();
+    this.pageNumber = ko.observable(1);
+    this.totalPages = ko.observable(0);
+    this.selectedProduct = ko.observable();
+    this.selectedProductName = ko.observable();
+    this.selection = ko.computed(() => {
+        const product = ko.unwrap(this.selectedProduct);
+        return product ? product.displayName : "Select API: products";
+    });
+  }
+
+  @Param()
+  public allowSelection: ko.Observable<boolean>;
+  
+  @Param()
+  public detailsPageUrl: ko.Observable<string>;
+
+  @OnMounted()
+  public async initialize(): Promise<void> {
+    const apiName = this.routeHelper.getApiName();
+
+    if (apiName) {
+      this.selectedApiName(apiName);
+      await this.loadPageOfProducts();
```

Comment:
Since `resetSearch()`  from the next line will call `loadPageOfProducts()` , is  this call needed here?